swap_meet/vendor.py

```diff
@@ -1,2 +1,57 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        self.inventory = inventory
+        self.inventory = self.inventory if inventory is not None else []
+
+    def add(self, item):
+
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item not in self.inventory: 
+            return False
+        else:
+            self.inventory.remove(item)
+            return item
+
+    def get_by_category(self, category):
+        item_by_category = []
+
+        for item in self.inventory:
+            if category == item.category:
+                item_by_category.append(item)
+        return item_by_category
+    
+    def swap_items(self,other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        else:
+            other_vendor.add(my_item)
+            other_vendor.remove(their_item)
+            self.add(their_item)
+            self.remove(my_item)
+            return True
+    
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        else:
+            return self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
```

Comment:
Nice function reuse. Though notice that due to the remove calls in `swap_items`, this will be O(n) where n is the lengths of the inventories, since we'll have to shift the whole remaining part of the inventories forward. By taking a different approach with `swap_items` or customizing our implementation here, we could do this swap in constant time.